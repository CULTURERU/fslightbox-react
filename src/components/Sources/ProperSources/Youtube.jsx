import React, { Component } from 'react';
import PropTypes from "prop-types";
import { getYoutubeVideoIDFromURL } from "../../../utils/SourceType/getYoutubeVideoIDFromURL";

class Youtube extends Component {
    componentDidMount() {
        this.props.fsLightbox.elements.sources[this.props.i].current.classList.remove('fslightbox-opacity-0');
        if (this.props.fsLightbox.sourcesData.isSourceAlreadyLoadedArray[this.props.i]) {
            return;
        }
        this.props.fsLightbox.sourcesData.sourcesDimensions[this.props.i] = {
            width: 1920,
            height: 1080,
        };
        this.props.onFirstSourceLoad();
    }

    render() {
        return (
            <iframe
                className="fslightbox-single-source fslightbox-opacity-0"
                ref={ this.props.fsLightbox.elements.sources[this.props.i] }
                src={
                    "https://www.youtube.com/embed/"
                    + getYoutubeVideoIDFromURL(this.props.fsLightbox.data.urls[this.props.i])
                    + '?enablejsapi=1'
                }
                allowFullScreen={ true }
                frameBorder="0"
            />
        );
    }
}

Youtube.propTypes = {
    fsLightbox: PropTypes.object.isRequired,
    i: PropTypes.number.isRequired,
    onFirstSourceLoad: PropTypes.func.isRequired,
};
export default Youtube;