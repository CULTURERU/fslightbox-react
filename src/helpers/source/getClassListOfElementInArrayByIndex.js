/**
 * @param { Array } arrayOfElements
 * @param { number } index
 * @return { DOMTokenList }
 */
export const getClassListOfElementInArrayByIndex = (arrayOfElements, index) => {
    return arrayOfElements[index].current.classList;
};