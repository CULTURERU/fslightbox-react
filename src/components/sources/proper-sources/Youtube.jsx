import React, { useEffect } from 'react';
import { getYoutubeVideoIdFromUrl } from "../../../helpers/source/getYoutubeVideoIdFromUrl";
import { PREFIX, SOURCE_CLASSES_NAMES } from "../../../constants/classes-names";

const Youtube = (
    {
        fsLightbox: {
            data: { sources },
            elements: { sources: sourcesElements },
            collections: {
                sourcesLoadsHandlers
            }
        },
        index,
    }
) => {
    useEffect(sourcesLoadsHandlers[index].handleLoad);

    return (
        <iframe
            className={ `${ SOURCE_CLASSES_NAMES } ${ PREFIX }youtube-iframe` }
            ref={ sourcesElements[index] }
            src={
                "https://www.youtube.com/embed/"
                + getYoutubeVideoIdFromUrl(sources[index])
                + '?enablejsapi=1'
            }
            allowFullScreen={ true }
        />
    );
};

export default Youtube;
