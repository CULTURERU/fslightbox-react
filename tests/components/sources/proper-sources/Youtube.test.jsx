import React from 'react';
import { mount, shallow } from "enzyme";
import Youtube from "../../../../src/components/sources/proper-sources/Youtube";
import { getYoutubeVideoIdFromUrl } from "../../../../src/helpers/source/getYoutubeVideoIdFromUrl";

const sourceController = {
    setIndex: () => {},
    setSourceWidth: () => {},
    setSourceHeight: () => {},
    normalLoad: () => {},
    initialLoad: () => {}
};
let fsLightbox = {
    data: {
        urls: ['https://www.youtube.com/watch?v=testId'],
    },
    sourcesData: {
        videosPosters: ['asdf'],
        isSourceAlreadyInitializedArray: []
    },
    elements: {
        sources: [{
            current: null
        }]
    },
    core: {
        sourceController: sourceController
    }

};

let youtube;

describe('ref to sources array in fsLightbox object', () => {
    beforeAll(() => {
        youtube = mount(<Youtube
            fsLightbox={ fsLightbox }
            index={ 0 }
        />);
    });

    it('should be equal to image', () => {
        expect(fsLightbox.elements.sources[0].current).toEqual(youtube.getDOMNode());
    });
});

describe('onload', () => {
    describe('initial load', () => {
        beforeAll(() => {
            sourceController.setIndex = jest.fn();
            sourceController.setSourceHeight = jest.fn();
            sourceController.setSourceWidth = jest.fn();
            sourceController.normalLoad = jest.fn();
            sourceController.initialLoad = jest.fn();
            // source is loaded for first time
            fsLightbox.sourcesData.isSourceAlreadyInitializedArray[0] = false;
            youtube = mount(<Youtube
                fsLightbox={ fsLightbox }
                index={ 0 }
            />);
        });

        it('should not call normalLoad', () => {
            expect(sourceController.normalLoad).not.toBeCalled();
        });

        it('should call setIndex with 0', () => {
            expect(sourceController.setIndex).toBeCalledWith(0);
        });

        it('should call setSourceWidth with 100', () => {
            expect(sourceController.setSourceWidth).toBeCalledWith(1920);
        });

        it('should call setSourceHeight with 150', () => {
            expect(sourceController.setSourceHeight).toBeCalledWith(1080);
        });

        it('should call initialLoad', () => {
            expect(sourceController.initialLoad).toBeCalled();
        });
    });


    describe('normal load', () => {
        beforeAll(() => {
            sourceController.setIndex = jest.fn();
            sourceController.setSourceHeight = jest.fn();
            sourceController.setSourceWidth = jest.fn();
            sourceController.normalLoad = jest.fn();
            sourceController.initialLoad = jest.fn();
            // source is loaded for first time
            fsLightbox.sourcesData.isSourceAlreadyInitializedArray[0] = true;
            youtube = mount(<Youtube
                fsLightbox={ fsLightbox }
                index={ 0 }
            />);
        });

        it('should not call setSourceWidth', () => {
            expect(sourceController.setSourceWidth).not.toBeCalled();
        });

        it('should not call setSourceHeight', () => {
            expect(sourceController.setSourceHeight).not.toBeCalled();
        });

        it('should not call initialLoad', () => {
            expect(sourceController.initialLoad).not.toBeCalled();
        });

        it('should call setIndex with 0', () => {
            expect(sourceController.setIndex).toBeCalledWith(0);
        });

        it('should call normalLoad', () => {
            expect(sourceController.normalLoad).toBeCalled();
        });
    });
});

describe('Video DOM', () => {
    beforeAll(() => {
        youtube = shallow(<Youtube fsLightbox={ fsLightbox } index={ 0 }/>);
    });

    describe('src', () => {
        it('should be equal to base embed youtube url + youtube id + js api enables', () => {
            expect(youtube.prop('src')).toBe("https://www.youtube.com/embed/testId?enablejsapi=1");
        });
    });

    describe('static DOM', () => {
        it('should match snapshot', () => {
            expect(youtube).toMatchSnapshot();
        });
    });
});



