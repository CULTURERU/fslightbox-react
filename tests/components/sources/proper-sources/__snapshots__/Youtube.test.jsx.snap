// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video DOM static DOM should match snapshot 1`] = `
<iframe
  allowFullScreen={true}
  className="fslightbox-source fslightbox-opacity-0 fslightbox-youtube-iframe"
  src="https://www.youtube.com/embed/testId?enablejsapi=1"
/>
`;
